src/dotnet/commands/dotnet-projectmodel-server/ProjectManager.cs

```diff
@@ -200,8 +200,10 @@ private bool ProcessMessage()
                     _configure.Value = message.Payload.GetValue("Configuration");
                     break;
                 case MessageTypes.RefreshDependencies:
+                    _refreshDependencies.Value = false;
+                    break;
                 case MessageTypes.RestoreComplete:
-                    _refreshDependencies.Value = 0;
+                    _refreshDependencies.Value = message.Payload.HasValues ? message.Payload.Value<bool>("Reset") : true;
```

Comment:
Here's the summary:

`RefreshDependencies` request: never clear cache forcefully.

`RestoreComplete` request: 

1. `Reset` property is not set - Clear cache;
2. `Reset` property is set to `true` - Clear cache;
3. `Reset` property is set to `false` - Do not clear cache.

So it's up to tooling to determine if the cache should be clear after package restore happened. By default the cache is cleared, which means performance hit.